mongoengine/connection.py

```diff
@@ -129,18 +143,16 @@ def get_connection(alias=DEFAULT_CONNECTION_NAME, reconnect=False):
                 conn_settings['slaves'] = slaves
                 conn_settings.pop('read_preference', None)
 
-        connection_class = MongoClient
         if 'replicaSet' in conn_settings:
             conn_settings['hosts_or_uri'] = conn_settings.pop('host', None)
-            # Discard port since it can't be used on MongoReplicaSetClient
+            # Discard port since it can't be used with replicaSet
             conn_settings.pop('port', None)
             # Discard replicaSet if not base string
             if not isinstance(conn_settings['replicaSet'], str):
                 conn_settings.pop('replicaSet', None)
-            connection_class = MongoReplicaSetClient
 
         try:
-            _connections[alias] = connection_class(**conn_settings)
+            _connections[alias] = MongoClient(**conn_settings)
```

Comment:
Calling out this change explicitly. https://pymongo.readthedocs.io/en/stable/migrate-to-pymongo4.html#mongoreplicasetclient says:
> Since PyMongo 3.0, MongoReplicaSetClient has been identical to [pymongo.mongo_client.MongoClient](https://pymongo.readthedocs.io/en/stable/api/pymongo/mongo_client.html#pymongo.mongo_client.MongoClient). Applications can simply replace MongoReplicaSetClient with [pymongo.mongo_client.MongoClient](https://pymongo.readthedocs.io/en/stable/api/pymongo/mongo_client.html#pymongo.mongo_client.MongoClient) and get the same behavior.

This is an example of a change that's easy to deploy everywhere as a separate PR.